odoo/cli/module.py

```diff
@@ -0,0 +1,152 @@
+import logging
+import textwrap
+from contextlib import contextmanager
+
+import odoo
+from odoo.cli.command import Command
+from odoo.modules.loading import force_demo
+from odoo.modules.module import get_module_path, initialize_sys_path
+from odoo.modules.registry import Registry
+from odoo.tools import config
+
+_logger = logging.getLogger(__name__)
+
+
+class Module(Command):
+    """ Install, Upgrade, Uninstall modules, install demo data """
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        subparsers = self.parser.add_subparsers(
+            dest='subcommand', required=True,
+            help='Subcommands help')
+
+        self.force_demo_parser = subparsers.add_parser(
+            'force-demo',
+            help="Install demo data (force)",
+            description="Install demonstration data (force)",
+        )
+        self.install_parser = subparsers.add_parser(
+            'install',
+            help="Install modules",
+            description="Install selected modules",
+        )
+        self.upgrade_parser = subparsers.add_parser(
+            'upgrade',
+            help="Upgrade modules",
+            description="Upgrade selected modules",
+        )
+        self.uninstall_parser = subparsers.add_parser(
+            'uninstall',
+            help="Uninstall modules",
+            description="Uninstall selected modules",
+        )
+
+        for parser in (
+            self.install_parser,
+            self.uninstall_parser,
+            self.force_demo_parser,
+            self.upgrade_parser,
+        ):
+            parser.add_argument(
+                '-c', '--config', dest='config',
+                help="use a specific configuration file")
+            parser.add_argument(
+                '-d', '--database', dest='db_name', default=None,
+                help="database name, connection details will be taken from the config file")
+
+        self.install_parser.add_argument(
+            'modules', nargs='+', metavar='MODULE',
+            help="modules to be installed")
+        self.install_parser.epilog = textwrap.dedent("""\
+            Before installing modules, an Odoo database needs
+            to be created and initialized on your PostgreSQL
+            instance, using the `db create` command:
+
+            $ odoo-bin db create <db_name>
```

Comment:
Currently `$ odoo-bin db create <db_name>` is not supported right? Maybe add the epilog later? Developers still need to use `odoo-bin -i` to initialize a database

Reply:
Ive made a separate PR, I will link it asap (courses atm)

Reply:
https://github.com/odoo/odoo/pull/209124 from mobile